src/regdesk/regdesk.js

```diff
@@ -0,0 +1,106 @@
+    <button id="printto_${index}" class="btn btn-success btn-lg" data-label-width="${printer.labelWidth}" data-label-height="${printer.labelHeight}" data-device-id="${index}">🖨</button>
+        <button class="btn btn-success dropdown-toggle dropdown-toggle-split" data-bs-toggle="dropdown" aria-expanded="false">
+            <span class="visually-hidden">Settings</span>
+        </button>
+        <ul class="dropdown-menu">
+            <li><a id="printtest_${index}" data-device-id="${index}" class="dropdown-item" href="#">
+                Print test page
+            </a></li>
+            <li><a id="autosense_${index}" data-device-id="${index}" class="dropdown-item" href="#">
+                AutoSense length
+            </a></li>
+            <li><a id="configprinter_${index}" data-device-id="${index}" class="dropdown-item" href="#">
+                Set label config
+            </a></li>
+            <li><a id="cutprinter_${index}" data-device-id="${index}" class="dropdown-item" href="#">
+                Cut label
+            </a></li>
+        </ul>
+    </div>
+</div>
+</div>
+</li>`;
+    document.getElementById("printerlist").appendChild(element);
+
+    document.getElementById(`printto_${index}`)
+        .addEventListener('click', async (e) => {
+            e.preventDefault();
+            let deviceId = e.target.dataset.deviceId;
+            printerMgr.printForm(deviceId);
+        });
+
+    document.getElementById(`autosense_${index}`)
+        .addEventListener('click', async (e) => {
+            e.preventDefault();
+            let deviceId = e.target.dataset.deviceId;
+            await printerMgr.printers[deviceId].configLabelWidth();
+            await printerMgr.printers[deviceId].setLabelHeightCalibration();
+        });
+
+    document.getElementById(`printtest_${index}`)
+        .addEventListener('click', async (e) => {
+            e.preventDefault();
+            let deviceId = e.target.dataset.deviceId;
+            await printerMgr.printers[deviceId].printTestPage();
+        });
+
+    document.getElementById(`cutprinter_${index}`)
+        .addEventListener('click', async (e) => {
+            e.preventDefault();
+            let deviceId = e.target.dataset.deviceId;
+            await printerMgr.printers[deviceId].addCmd("C").print();
+        });
+}
+
+////////////////////////////////
+// End most of the test code, though the rest of this will change too.
+
+
+var printerMgr;
+
+document.addEventListener('readystatechange', async () => {
+    if (document.readyState === "complete") {
+
+        const printerList = document.getElementById("printerlist");
+        printerMgr = new PrinterManager();
+
+        document.getElementById("addprinter").addEventListener('click', async () => {
+            await printerMgr.printerPair();
+            resetPrinterButtons(printerMgr);
+        });
+
+        navigator.usb.addEventListener('connect', async (e) => printerMgr.printerConnect(e));
```

Comment:
This is so cool learning about usb interaction in the browser! Thank you for taking the initiative to learn the hard parts and make it look easy for me :D.